docs/python/data-science-tutorial.md

```diff
@@ -0,0 +1,250 @@
+
+1. A common first step to training a model is to divide up the dataset into training and validation data. This allows you to use a portion of the data to train the model and a portion of the data to test the model. If you used all your data to train the model, you wouldn’t have a way to estimate how well it would actually perform against data the model has not yet seen. A benefit of the scikit-learn library is that it provides a method specifically for splitting a dataset into training and test data.
+
+   Add and run a cell with the following code to the notebook to split up the data.
+
+   ```python
+   from sklearn.model_selection import train_test_split
+   x_train, x_test, y_train, y_test = train_test_split(data[['sex','pclass','age','relatives','fare']], data.survived, test_size=0.2, random_state=0)
+   ```
+
+1. Next, you'll normalize the inputs such that all features are treated equally. For example, within the dataset the values for age range from ~0-100, while gender is only a 1 or 0. By normalizing all the variables, you can ensure that the ranges of values are all the same. Use the following code in a new code cell to scale the input values.
+
+   ```python
+   from sklearn.preprocessing import StandardScaler
+   sc = StandardScaler()
+   X_train = sc.fit_transform(x_train)
+   X_test = sc.transform(x_test)
+   ```
+
+1. There are a number of different machine learning algorithms that you could choose from to model the data and scikit-learn provides support for a number of [them](https://scikit-learn.org/stable/user_guide.html), as well as a [chart](https://scikit-learn.org/stable/tutorial/machine_learning_map/index.html) to help select the one that’s right for your scenario. For now, use the [naïve bayes algorithm](https://scikit-learn.org/stable/modules/naive_bayes.html), a common algorithm for classification problems. Add a cell with the following code to create and train the algorithm.
+
+   ```python
+   from sklearn.naive_bayes import GaussianNB
+   model = GaussianNB()
+   model.fit(X_train, y_train)
+   ```
+
+1. With a trained model, you can now try it against the test data set that was held back from training. Add and run the following code to predict the outcome of the test data and calculate the accuracy of the model.
+
+   ```python
+   from sklearn import metrics
+   predict_test = model.predict(X_test)
+   print(metrics.accuracy_score(y_test, predict_test))
+   ```
+
+   ![Running the trained model against test data](images/data-science-tutorial/jupyter-cell-05.png)
+
+   Looking at the result of the test data, you’ll see that the trained algorithm had a ~75% success rate at estimating survival.
+
+## (Optional) Use a neural network to increase accuracy
```

Comment:
TODO Jeffrey: Change the code to use PyTorch

Reply:
Per conversation, keeping as Keras/Tensorflow for now.